website_product_brand/controllers/main.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2012-Today Serpent Consulting Services Pvt. Ltd.
+#                                     (<http://www.serpentcs.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
+
+import werkzeug
+from openerp import http
+from openerp.http import request
+import openerp.addons.website_sale.controllers.main
+from openerp import SUPERUSER_ID
+from openerp.addons.website.models.website import slug
+from openerp.addons.website_sale.controllers.main import table_compute
+from openerp.addons.website_sale.controllers.main import QueryURL
+PPG = 20
+PPR = 4
+
+
+class WebsiteSale(openerp.addons.website_sale.controllers.main.website_sale):
+
+    @http.route(['/shop',
+                 '/shop/page/<int:page>',
+                 '/shop/category/<model("product.public.category"):category>',
+                 """/shop/category/<model("product.public.category"):category>
+                 /page/<int:page>""",
+                 '/shop/brands'],
+                type='http',
+                auth='public',
+                website=True)
+    def shop(self, page=0, category=None, brand=None, search='', **post):
```

Comment:
This method is way too long and doesn't call the `super` implementation, breaking inheritance.
Also, it seems to be identical except for lines 98-105 that add the brand condition to the domain.
The `brand` parameter in never captured in the route string, where is it supposed to come from?